main.tf

```diff
@@ -39,10 +39,20 @@ resource "aws_route53_zone" "zone" {
 locals {
   records_expanded = [
     for record in var.records : merge({
-      record_id = join("-", compact([lower(record.type), lower(record.name)]))
-      ttl       = null
-      records   = null
-      alias     = null
+      record_id = join("-", compact([lower(record.type), lower(record.name),
+        try(
+          "${lower(record.set_identifier)}-${lower(record.weight)}",                                    # weighted routing policy
+          "${lower(record.set_identifier)}-${lower(record.failover)}-${lower(record.health_check_id)}", # failover
+          null
+        ),
+      ]))
+      ttl             = null
+      records         = null
+      alias           = null
+      set_identifier  = null
+      weight          = null
+      failover        = null
+      health_check_id = null
```

Comment:
merge will remove null values anyway.. so there is no point adding them here

Reply:
<img width="450" alt="Screen Shot 2020-02-18 at 17 40 24" src="https://user-images.githubusercontent.com/930643/74757066-c9e37600-5275-11ea-8ad5-4283aa8afa9b.png">

Reply:
wasn't that introduced by you ? :)

Reply:
;) yes... found this out when debugging the provider issue with alias... so can be removed.. just forgot to clean up afterwards as it seems... my bad..

Reply:
learning curve for the win!

Reply:
i assume we also do this in other modules... needs fixing i guess

Reply:
doesn't look like it removes the null values.. if i remove the lines, the actual implementation breaks.

<img width="573" alt="Screen Shot 2020-02-18 at 17 55 01" src="https://user-images.githubusercontent.com/930643/74758567-cfda5680-5277-11ea-8f7c-8c09216de789.png">

Reply:
@mariux